osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/config/ThrottlingClientConfig.java

```diff
@@ -54,10 +71,27 @@ public ThrottlingClient throttlingClient() {
                 this.configurationName, this.configurationMaxConcurrency),
             this.throttlingServiceUrl,
             this.timeout);
-    throttlingClient.register();
+
+    this.registerThrottlingClient(throttlingClient);
     return throttlingClient;
   }
 
+  protected void registerThrottlingClient(final ThrottlingClient throttlingClient) {
+    try {
+      log.info("Try to register the client to the throttling service");
+      throttlingClient.register();
+    } catch (final ResourceAccessException resourceAccessException) {
```

Comment:
Wouldn't it be better to wrap this for any kind of exception? Something similar goes for whether any exception should ever be thrown from this method (like the `BeanInitializationException` below.

The reason I comment about this, is that it is not clear to me what the consequence of any exception being thrown from the lazy bean method `throttlingClient` would be for the chances of the protocol adapter to be able to use throttling (and with that function properly).
If the bean in such case never will be initialized, the protocol adapter should no longer be running.
If the bean will be initialized on a later call (not sure how the Spring proxying handles this) there is another question of whether the code calling the bean method handles the exception in an appropriate manner.
Such code never had to deal with issues in `throttlingClientConfig.throttlingClient()` with an autowired `throttlingClientConfig`, because the application would not have started if this caused errors.

Reply:
Agree, i think this exception should not been thrown at all, refactored it

Reply:
As `ResourceAccessException` is not the only exception that can possibly be thrown from `throttlingClient.register()`, and given the earlier reasoning, we don't want any exception escaping here, would it not be better to catch the more generic `Exception` here?

Reply:
We could do that, but i personally prefer to be as specific as possible.

Reply:
I know that it usually is a good idea to use that preference as a guideline, but in this specific case I understand we do not want **any** exception to pop up from the bean creation, and if so, we had better make sure that no exception escapes by catching them all. (As far as I see it, we do not have any better way to deal with any specific exception sub types, other than to keep trying to register.)

What could happen if this only catches `ResourceAccessException`? If `throttlingClient.register` throws `RestClientException` for instance, the bean method throws an exception.
What would be the result of that? That the protocol adapter remains unusable until somehow it is detected that it does not process any requests? If you go down the road of listing all specific exceptions that are currently thrown form any path in `throttlingClient.register` then you would set yourself up for trouble at some later time with code changes outside of the scope of this `ThrottlingClientConfig`.